### PR TITLE
perf: Improve Survey Results Memory Usage

### PR DESCRIPTION
**Pull request by ThomasGalesloot**
# Pull Request Key Information

## Reason for change

During the memory profiling, the results of surveys were found to be taking up over 70% of the memory allocated during simulation.

## What was changed

This change reduces the memory usage of survey results, resulting an approximately 97% decrease in memory allocated during simulation in the case being tested.

## Intended Purpose

Lower memory usage during simulation.

## Level of version change required

Patch

## Testing Completed

Manual testing with memory profiling - See Sample results here: 
[memory_profiling_results.zip](https://github.com/user-attachments/files/16286354/memory_profiling_results.zip)

All unit tests passing:
[unit_test_results.txt](https://github.com/user-attachments/files/16286272/unit_test_results.txt)

All End-to-End tests passing: 
[V4_simple_non_repairable_emissions_df7cdf012ba7c4aef1dbd54e02f1f18bd9b0bfa3.log](https://github.com/user-attachments/files/16286269/V4_simple_non_repairable_emissions_df7cdf012ba7c4aef1dbd54e02f1f18bd9b0bfa3.log)
[V4-simple_repairable_emissions_df7cdf012ba7c4aef1dbd54e02f1f18bd9b0bfa3.log](https://github.com/user-attachments/files/16286270/V4-simple_repairable_emissions_df7cdf012ba7c4aef1dbd54e02f1f18bd9b0bfa3.log)

## Target Issue

N/A

## Additional Information

N/A

### CODE REVIEW

File: LDAR_Sim/src/constants/output_file_constants.py
Comment (SEAJang) [inline LDAR_Sim/src/constants/output_file_constants.py:1]: EMIS_COMP_ESTIMATION_OUTPUT_COLUMNS is no longer being used anywhere else. Please remove.

File: LDAR_Sim/src/programs/method.py [hunk 1]
```diff
@@ -277,7 +277,7 @@ def deploy_crews(self, workplan: Workplan, weather, daylight) -> CrewDeploymentS
             # indicate that the particular site needs to be requeued with higher priority
             incompleteSurveys.append((survey_report, survey_plan))
             if survey_report.survey_complete:
-                self._site_survey_reports.append(survey_report)
+                survey_report.follow_up_method = self.get_follow_up_method_name()
```
Comment (SEAJang) [inline LDAR_Sim/src/programs/method.py:280]: This always returns "N/A". I don't believe it's being used anywhere.

File: LDAR_Sim/src/scheduling/schedule_dataclasses.py [hunk 2]
```diff
@@ -101,8 +121,39 @@ def to_report_summary(self, expand: bool = False):
             eca.SURVEY_COMPLETION_DATE: self.survey_completion_date,
             eca.SURVEY_START_DATE: self.survey_start_date,
             eca.METHOD: self.method,
+            eca.FU_METHOD: self.follow_up_method,
```
Comment (SEAJang) [inline LDAR_Sim/src/scheduling/schedule_dataclasses.py:124]: Can we double check that this is needed and being used?

File: LDAR_Sim/src/constants/output_file_constants.py [hunk 3]
```diff
@@ -268,6 +268,7 @@ class EMIS_DATA_COL_ACCESSORS:
     SITE_MEASURED = "Site Measured"
     SITE_TYPE = "Site Type"
     LAST_SURVEY_DATE = "Last Survey Date"
+    FU_METHOD = "follow_up_method"
```
Comment (SEAJang) [inline LDAR_Sim/src/constants/output_file_constants.py:271]: Double check that this is needed

**Review by SEAJang — CHANGES_REQUESTED** [review]
Great job. 
Can you look into if the follow-up method is even needed in code? 
I've done a quick look through and I think it's code that's just not needed. 
Otherwise, there's a little bit of cleanup as well.

File: LDAR_Sim/src/ldar_sim_run.py
[the remark this answers is not in the document]
Comment (ThomasGalesloot) [reply LDAR_Sim/src/ldar_sim_run.py:1]: Lets do this in a separate commit. Can you make an Issue for this

File: LDAR_Sim/src/programs/method.py (hunk 1 above)
Comment (ThomasGalesloot) [reply LDAR_Sim/src/programs/method.py:280]: You are correct - Will remove

**Comment by ThomasGalesloot** [comment]
New unit test results:
[unit_test_results.txt](https://github.com/user-attachments/files/16288024/unit_test_results.txt)


New End-to-End test results: 
[V4_simple_non_repairable_emissions_c67e910bf3e029273da17ce6bfa5a0ae997f9839.log](https://github.com/user-attachments/files/16288022/V4_simple_non_repairable_emissions_c67e910bf3e029273da17ce6bfa5a0ae997f9839.log)
[V4-simple_repairable_emissions_c67e910bf3e029273da17ce6bfa5a0ae997f9839.log](https://github.com/user-attachments/files/16288023/V4-simple_repairable_emissions_c67e910bf3e029273da17ce6bfa5a0ae997f9839.log)

**Review by SEAJang — APPROVED** [review]

**Pull request merged**